prefect_dbt/cloud/runs.py

```diff
@@ -0,0 +1,185 @@
+
+    Example:
+        List artifacts of a dbt Cloud job run:
+        ```python
+        from prefect import flow
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import list_dbt_cloud_run_artifacts
+
+        @flow
+        def list_artifacts_flow():
+            credentials = DbtCloudCredentials(api_key="my_api_key", account_id=123456789)
+
+            return list_dbt_cloud_run_artifacts(
+                dbt_cloud_credentials=credentials,
+                run_id=42
+            )
+
+        list_artifacts_flow()
+        ```
+    """  # noqa
+    try:
+        async with dbt_cloud_credentials.get_administrative_client() as client:
+            response = await client.list_run_artifacts(run_id=run_id, step=step)
+    except HTTPStatusError as ex:
+        raise DbtCloudListRunArtifactsFailed(extract_user_message(ex)) from ex
+    return response.json()["data"]
+
+
+@task(
+    name="Get dbt Cloud job artifact",
+    description="Fetches an artifact from a completed run.",
+    retries=3,
+    retry_delay_seconds=10,
+)
+async def get_dbt_cloud_run_artifact(
+    dbt_cloud_credentials: DbtCloudCredentials,
+    run_id: int,
+    path: str,
+    step: Optional[int] = None,
+):
+    """
+    A task to get an artifact generated for a completed run.
+
+    Args:
+        dbt_cloud_credentials: Credentials for authenticating with dbt Cloud.
+        run_id: The ID of the run to list run artifacts for.
+        path: The relative path to the run artifact (e.g. manifest.json, catalog.json,
+            run_results.json)
+        step: The index of the step in the run to query for artifacts. The
+            first step in the run has the index 1. If the step parameter is
+            omitted, then this method will return the artifacts compiled
+            for the last step in the run.
+
+    Returns:
+        The contents of the requested manifest.
+
+    Example:
+        Get an artifact of a dbt Cloud job run:
+        ```python
+        from prefect import flow
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import get_dbt_cloud_run_artifact
+
+        @flow
+        def get_artifact_flow():
+            credentials = DbtCloudCredentials(api_key="my_api_key", account_id=123456789)
+
+            return get_dbt_cloud_run_artifact(
+                dbt_cloud_credentials=credentials,
+                run_id=42,
+                path="manifest.json"
+            )
+
+        get_artifact_flow()
+        ```
+    """  # noqa
+
+    try:
+        async with dbt_cloud_credentials.get_administrative_client() as client:
+            response = await client.get_run_artifact(
+                run_id=run_id, path=path, step=step
+            )
+    except HTTPStatusError as ex:
+        raise DbtCloudGetRunArtifactFailed(extract_user_message(ex)) from ex
+    return response.json()
```

Comment:
Not returning the json I expect it to. 

Input
```python
import os

from prefect import flow
from prefect_dbt.cloud import DbtCloudCredentials
from prefect_dbt.cloud.runs import get_dbt_cloud_run_artifact

@flow
def get_artifact_flow():
    credentials = DbtCloudCredentials(api_key=os.getenv('DBT_TOKEN'), account_id=16173)
    artifact = get_dbt_cloud_run_artifact(
        dbt_cloud_credentials=credentials,
        run_id=66585620,
        path="run_results.json"
    )
    print(artifact)
    return artifact

get_artifact_flow()
```

Output
```shell
23:42:36.785 | INFO    | prefect.engine - Created flow run 'honest-otter' for flow 'get-artifact-flow'
23:42:36.786 | INFO    | Flow run 'honest-otter' - Using task runner 'ConcurrentTaskRunner'
23:42:36.878 | INFO    | Flow run 'honest-otter' - Created task run 'Get dbt Cloud job artifact-d6d62cbd-0' for task 'Get dbt Cloud job artifact'
PrefectFuture('Get dbt Cloud job artifact-d6d62cbd-0')
23:42:37.687 | INFO    | Task run 'Get dbt Cloud job artifact-d6d62cbd-0' - Finished in state Completed()
23:42:37.711 | INFO    | Flow run 'honest-otter' - Finished in state Completed('All states completed.')
```

Reply:
I think you'll get the output that you expect if you print `artifact.result()`. In this code, `artifact` is a `PrefectFuture` which will have the result of the task once execution completes. Calling `.result()` on a `PrefectFuture` waits for execution to complete and then returns the result.